kubernetes/host-logs-geneva.yaml

```diff
@@ -1,85 +1,108 @@
-        hostProcess: true
-        runAsUserName: "NT AUTHORITY\\SYSTEM"
-    hostNetwork: true
-    containers:
-     - name: host-logs-windows
-       image: "VALUE_IMAGE_NAME"
-       imagePullPolicy: IfNotPresent
-       resources:
-        limits:
-         cpu: "1"
-         memory: 2Gi
-       env:
-        - name: AKS_CLUSTER_NAME
-          value: "VALUE_AKS_CLUSTER_NAME"
-        - name: AKS_REGION
-          value: "VALUE_AKS_RESOURCE_REGION_VALUE"
-        - name: HOSTNAME
-          valueFrom:
-             fieldRef:
-               fieldPath: spec.nodeName
-       volumeMounts:
-        - mountPath: \etc\config\settings
-          name: settings-vol-config
-          readOnly: true
-       livenessProbe:
-          exec:
-            command:
-              - cmd
-              - /c
-              - .\opt\hostlogswindows\scripts\cmd\livenessprobe.exe
-              - ".\\etc\\hostlogswindows\\filesystemwatcher.txt"
-    affinity:
-     nodeAffinity:
-       requiredDuringSchedulingIgnoredDuringExecution:
-         nodeSelectorTerms:
-         - matchExpressions:
-           - key: kubernetes.io/os
-             operator: In
-             values:
-             - windows
-    tolerations:
-    - key: "CriticalAddonsOnly"
-      operator: "Exists"
-    - operator: "Exists"
-      effect: NoExecute
-    - operator: "Exists"
-      effect: NoSchedule
-    - operator: "Exists"
-      effect: PreferNoSchedule
-    volumes:
-      - name: settings-vol-config
-        configMap:
-          name: container-azm-ms-agentconfig
-          optional: true
\ No newline at end of file
+  template:
+    metadata:
+      labels:
+        component: geneva-agent-windows
+        tier: node-win
+      annotations:
+        agentVersion: 45.13.1
+        dockerProviderVersion: 18.0.1-0
+        schema-versions: v1
+    spec:
+      dnsConfig:
+        options:
+          - name: ndots
+            value: '3'
+      securityContext:
+        windowsOptions:
+          hostProcess: true
+          runAsUserName: NT AUTHORITY\SYSTEM
+      hostNetwork: true
+      containers:
+        - name: host-logs-windows
+          image: VALUE_CONTAINER_IMAGE
+          imagePullPolicy: IfNotPresent
+          resources:
+            requests:
+              cpu: '0.25'
+              memory: 250Mi
+            limits:
+              cpu: '0.5'
+              memory: 1Gi
+          env:
+            - name: AKS_RESOURCE_ID
+              value: VALUE_AKS_RESORUCE_ID
+            - name: AKS_CLUSTER_NAME
+              value: VALUE_AKS_CLUSTER_NAME
+            - name: AKS_REGION
+              value: VALUE_AKS_RESOURCE_REGION_VALUE
+            - name: HOSTNAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - mountPath: \etc\config\settings
+              name: settings-vol-config
+              readOnly: true
+          startupProbe:
+            initialDelaySeconds: 180
```

Comment:
180 seconds might be a bit long for the initial delay. I wouldn't expect it to take that long on a normal startup. The sooner we detect that it has started, the sooner we can switch to the liveness probe and start watching for config map changes.

And maybe increase the failure threshold to account for this change, so we don't fail after 3 minutes if it does take a while to start up

Reply:
The default failure threshold is 3 so I saw that was a good starting point as we haven't hit the issue after x amount of time at startup. If we encounter this being a problem, we can always bump this up.

The initalDelaySeconds is carried over from the liveness probe and since ama-logs-windows waits 180 seconds to check anything.

Reply:
A long initial delay makes sense when we have just a livenessprobe. If we start the livenessprobe too early it would think there's a problem and restart the pod

The purpose of the startup probe is a replacement for the initialdelay to determine when we're ready to start the livenessprobe. If it runs before we're ready that's ok – it will just run again. Ideally, we can detect that our pod started up correctly and switch over to the liveness probe as soon as possible. Basically, there's no benefit to a long initial delay, but a short initial delay means we can detect problems and/or configmap changes sooner

Reply:
Took it out